### DISCUSSION TITLE
How to ref a schema that was not defined by typebox?

### DISCUSSION THREAD

**Discussion by OmarAlkadri** [feathersjs/feathers#3360] [Q&A]
How can I create a userID whose reference is the same as users

i use feathersjs V5 and typeBox

export const usersSchema = Type.Object(
{
_id: ObjectIdSchema(),
email: Type.String(),
date: Type.String({ format: 'date' }),
password: Type.Optional(Type.String()),
userID: ObjectIdSchema(),

user: Type.Ref(usersSchema),
},
{ $id: 'Users', additionalProperties: false }
)
Block-scoped variable 'usersSchema' used before its declaration.ts(2448)
users.schema.ts(13, 14): 'usersSchema' is declared here.

**Comment by OmarAlkadri** [comment]
@daffl @marshallswain @vonagam @FossPrime @ekryski @DaddyWarbucks 

Hello, we have been using feathersjs for a long time and now we want to move to feathersjs V 5, but I am now creating an experiment project and I am facing a lot of problems.
The most important of which is that I cannot create a reference from the same service
I am having problems sending parameters to the hooks I created
Can you please suggest me some solutions?

**Comment by daffl** [comment]
Recursive schemas in TypeBox should be possible like this:

```ts
export const usersSchema = Type.Object(
{
_id: ObjectIdSchema(),
email: Type.String(),
date: Type.String({ format: 'date' }),
password: Type.Optional(Type.String()),
userID: ObjectIdSchema(),

user: Type.Ref(() => usersSchema),
},
{ $id: 'Users', additionalProperties: false }
)
```

**Reply by jeevasuriyan** [reply]
@daffl 
Hi i used this recursive but it is not working for me, what should i do it is showing this error 

```Block-scoped variable 'modelsSchema' used before its declaration.ts(2448)
models.schema.ts(14, 14): 'modelsSchema' is declared here.```
